docs/changes/units/14042.feature.rst

```diff
@@ -0,0 +1,3 @@
+Add support for degrees Celsius for FITS. Parsing "deg C" and "Celsius" is now
+supported and astropy will output "Celsius" into FITS. Beware that compound units
```

Comment:
I'd swap "Celsius" and "deg C" to make clear the former is preferred, and perhaps be explicit about that: "Note that "deg C" is only provided for compatibility with existing FITS files, as it does not conform to the normal unit standard, where this should be read as "degree * Coloulomb". Indeed, compound units ..."